### PR TITLE
feat: Allowed to indicate the source of nuget package and whether it is a pre-release version.

### PR DESCRIPTION
**Pull request by workgroupengineering**
Added source and prerelease optional parameters at WithNuGet extension method

### CODE REVIEW

File: src/BenchmarkDotNet/Jobs/NugetReference.cs
Comment (adamsitnik) [inline src/BenchmarkDotNet/Jobs/NugetReference.cs:22]: nit: empty line

```suggestion
            PackageSource = source;
            Prerelease = prerelease;
```

File: src/BenchmarkDotNet/Toolchains/DotNetCli/DotNetCliCommand.cs
Comment (adamsitnik) [inline src/BenchmarkDotNet/Toolchains/DotNetCli/DotNetCliCommand.cs:220]: nits:
* typo => commadBuilder => comma**n**dBuilder 
* `boolean == false` we are rather using `!boolean` (style consistency with the rest of the repo)

```suggestion
            var commandBuilder = new StringBuilder();
            commadBuilder.Append("add package ");
            commadBuilder.Append(reference.PackageName);
            if (!string.IsNullOrWhiteSpace(reference.PackageVersion))
            {
                commadBuilder.Append(" -v ");
                commadBuilder.Append(reference.PackageVersion);
            }
            if (reference.PackageSource != null)
            {
                commadBuilder.Append(" -s ");
                commadBuilder.Append(reference.PackageSource);
            }
            if (reference.Prerelease)
            {
                commadBuilder.Append(" --prerelease");
            }
            return commadBuilder.ToString();
        }
```

File: src/BenchmarkDotNet/Jobs/JobExtensions.cs
Comment (adamsitnik) [inline src/BenchmarkDotNet/Jobs/JobExtensions.cs:351]: I think that instead of having two overloads with optional parameters we should have a single method with all optional parameters at once place:

```diff
- public static Job WithNuGet(this Job job, string packageName, string packageVersion, Uri source = null, bool prerelease = false)
- public static Job WithNuGet(this Job job, string packageName, Uri source = null, bool prerelease = false)
+ public static Job WithNuGet(this Job job, string packageName, string packageVersion = null, Uri source = null, bool prerelease = false)
```

and `null` for `packageVersion` should mean no package version specified (currently we have `string.Empty` for `packageVersion` and `null` for `source` (consistency)

**Review by adamsitnik — COMMENTED** [review]
Hi @ workgroupengineering !

Big thanks for your PR! Overall it looks good, but please address my feedback before I hit the merge button.

Thanks,
Adam

**Comment by workgroupengineering** [comment]
Thanks for review

File: src/BenchmarkDotNet/Jobs/JobExtensions.cs
Comment (adamsitnik) [inline src/BenchmarkDotNet/Jobs/JobExtensions.cs:332]: The `packageVersion` parameter is also optional now.

```suggestion
        /// <param name="packageVersion">(optional)The NuGet package version</param>
```

**Review by adamsitnik — COMMENTED** [review]
@workgroupengineering thank you for applying the suggestions! Could you please take a look at the failing test?

```log
 BenchmarkDotNet.Tests.Configs.JobTests.WithNuGet

Assert.Collection() Failure
Collection: [AutoMapper 7.0.1, Newtonsoft.Json]
Error during comparison of item at index 1
Inner exception: Assert.Equal() Failure
        Expected: Newtonsoft.Json
        Actual:   Newtonsoft.Json
```

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @workgroupengineering !

**Pull request merged**
